app/models/order.rb

```diff
@@ -24,7 +24,9 @@ class Order < ApplicationRecord
 
   ### CALLBACKS ###
 
+  before_save :check_placed_on, if: %i[cancelled? status_changed?]
   after_commit :send_confirmation_email, if: :received?
+  after_update_commit :increase_inventory, if: %i[cancelled? status_previously_changed?]
```

Comment:
no need to create a new callback if you want to call this just with single method `cancel`
You can move this method logic in the same method.

Reply:
Done

Reply:
Done